src/light.cpp

```diff
@@ -29,33 +29,55 @@ static u8 light_LUT[LIGHT_SUN + 1];
 // The const ref to light_LUT is what is actually used in the code
 const u8 *light_decode_table = light_LUT;
 
+struct LightingParams {
+	float a, b, c; // polynomial coefficients
+	float boost, center, sigma; // normal boost parameters
+	float gamma;
+};
+
+static LightingParams params;
+
+float decode_light_f(float x)
+{
+	x = rangelim(x, 0.0f, 1.0f);
+	float brightness = params.a * ipow(x, 3) + params.b * sqr(x) + params.c * x;
```

Comment:
With a total of three multiplications `(((params.a * x + params.b) * x + params.c) * x`

Reply:
Oh, Horner's method... okay.

Reply:
Whoops, a typo... (extra `(` )

Reply:
Suggested this solution in order to make `ipow` superfluous and to increase the execution speed a tiny bit. Sorry for the typo, but you got the idea. :)